docs/design/queue/queue-priority-design.md

```diff
@@ -0,0 +1,418 @@
+   can basically determine the order.
+3. Priority plugin in the end. In this case, Priority basically cannot play a role, because the previous plugin(drf and proportion)
+   can basically determine the queue order, unless the `Share` of queue  calculated by DRF or Proportion is 0 or 1.
+
+### Better QueueOrderFn -> QueueScoreOrderFn
+The above configuration is suitable for whose only consider on one or two kinds of queue sorting requirements, or one of then is
+the main sorting strategy, and the other is the secondary sorting strategy.
+
+Here, propose a strategy that takes into account all queue sorting in three plugin, similar to calculating `NodeScore`,
+use three plugin to calculate the score of each queue. **Top score has highest-priority**. user can customize weight of three plugins
+to meet their preference for a certain priority.
+
+* Queue's Score Of Priority
+
+  `Score = (PriorityClass of queue - Minimum PriorityClass)/(Maximum PriorityClass - Minimum PriorityClass) * Weight of Priority Plugin`
+
+* Queue's Score Of DRF
+
+  `Score = (1 - node.attr.share/node.weight) * Weight of DRF Plugin`
+
+* Queue's Score Of Proportion
+
+  `Score = (1 - queue.share) * Weight of Proportion Plugin`
+
+`Total Score = Priority's Score + DRF's Score + Proportion's Score`
+
+#### QueueScoreOrderFn example：
+```
+Maximum PriorityClass = 100
+Minimum PriorityClass = 0
+Priority Weight = 1
+DRF Weight = 1
+Proportion Weight = 1
+```
+|        | Priority | DRF  Share | Proportion Share | Priority Score | DRF Score | Proportion Score | Total Score |
+|--------|----------|------------|------------------|----------------|-----------|------------------|-------------|
+| QueueA | 40       | 0.3        | 1.2              | 0.4            | 0.7       | -0.2             | 0.9         |
+| QueueB | 80       | 0.4        | 0.5              | 0.8            | 0.6       | 0.5              | 1.9         |
+| QueueC | 0        | 0.5        | 0.3              | 0              | 0.5       | 0.7              | 1.2         |
+
+QueueScoreOrder： **Queue B > Queue C > Queue A**
+
+### Impact priority of job
+
+Due to the Priority of queue, therefore, **if priority not setting in job, the job priority will inherit the priority of queue**.
+
+*Through the idea of QueueScore, here is a reflection：*
+
+At present, all of enqueue、 allocate or reclaim actions are sort the jobs in the queue after the queue is sorted.
+If the following situation occur.
+
+There are two queue:
+
+Queue A(priority 10) has A1 Job(default priority 10), A2 Job(priority 12).
+
+Queue B(priority 8) has B1 Job(default priority 8), B2 Job(priority 13).
+
+Current logic of the three actions are as follows:
+* A Queue
+    * A2 Job
+    * A1 Job
+* B Queue
+    * B2 Job
+    * B1 Job
+
+Try to think another way, if we think job is the base unit in scheduler, priority of job maybe more important than priority of queue.
+
+New logic of the three actions are as follows:
+* B2 Job
+* A2 Job
+* A1 Job
+* B1 Job
```

Comment:
Kindly ask how to ensure the priority fairness among queues? For example, if this rule works, users under Queue B submit all jobs with the highest priority to make sure its jobs can stand in the head of the to-be-scheduled queue for the users know their queue has a relative lower priority.

Reply:
This is just a thought I put forward from the logic of queue scoring. If we judge the overall priority according to the job as a scheduling unit, whether it is such a result without considering the queue. The thinking here has nothing to do with this design. Maybe there will be requirements in the future. For example, the priority, utilization rate or dominant resource utilization rate of the queue will only be a factor affecting the job sorting, and the overall scheduling will be determined by the job sorting. no longer related to the queue.

Reply:
@william-wang @shinytang6 Glad to see your viewpoints about the fairness and the throughout for specified users.